### ISSUE TITLE
LSTM not predicting time series correctly

### ISSUE THREAD

**Issue by tgs266** [tflearn/tflearn#374]
So I am trying to predict the stock market, and before you say anything about the stock market being impossible to predict, I predicted Twitter's stock on Friday down to a cent. But I was looking to move my network over to Tflearn from Keras. Here is both Keras and Tflearn network code:

Keras

           `model = Sequential()
            model.add(LSTM(layers[0], input_dim=look_back, return_sequences = True))
            model.add(Dropout(0.2))

            model.add(LSTM(layers[1], return_sequences=True))
            model.add(Dropout(0.2))

            model.add(LSTM(layers[2], return_sequences=False))
            model.add(Dropout(0.2))

            model.add(Dense(output_dim=1))
            model.add(Activation("linear"))

            start = time.time()
            model.compile(loss="mse", optimizer="adam")
            end = time.time()
            print ("Build Time: ", end - start)`

Tflearn

           `start = time.time()
            model = input_data(shape=[None, trainX.shape[1]])
            model = tflearn.embedding(model, input_dim=trainX.shape[1], output_dim=look_back)

            model = lstm(model, layers[0], return_seq=True, activation="sigmoid")
            model = dropout(model, 0.2)

            model = lstm(model, layers[1], return_seq=True)
            model = dropout(model, 0.2)
            
            model = lstm(model, layers[2], return_seq=False)
            model = dropout(model, 0.2)

            model = fully_connected(model, 1)
            model = activation(model, activation="linear")

            model = regression(model, optimizer = "adam", learning_rate=0.01, loss='mean_square')
            model = tflearn.DNN(model, tensorboard_verbose=0)
            end = time.time()
            print ("Build Time: ", end - start)`

With Keras, I was able to get great results, it was just very slow. But tflearn seems to operate faster, but my predictions range between 150, and -25, when changing around different parameters. But I kept 100 neurons on layer 1, 200 on layer 2, 300 on layer 3, and changed around things like activation, optimizer, and learning rate. But I can't get results anything near as close to keras.

I am happy to provide more code if necessary, but the code is the same between libraries, and the only change is the different code needed for each library.  What should I do?

**Comment by aymericdamien** [comment]
What about learning rate? is Keras also using 0.01? This may be an important parameter.
Otherwise, some noticeable differences:
- Initialization functions for layers, be sure you use same ones.
- TFLearn clip gradients by default, you may disable that `tflearn.DNN(..., clip_gradients=0.)`.

**Comment by tgs266** [comment]
@aymericdamien I'll try it out. Thanks!

**Comment by tgs266** [comment]
@aymericdamien I changed the learning rate, clipping, and I adjusted the Adam optimizer to match the Keras one, but I can't get it to go lower than 35 as a prediction, when in Keras I can get it as close as 1 dollar. Anything else you might recommend? I really want to switch over to Tflearn because it is so much faster, but I don't know if I can if I can't get this to work.

**Comment by aymericdamien** [comment]
Is there a reason why you use activation sigmoid here?
```
model = lstm(model, layers[0], return_seq=True, activation="sigmoid")
```
Can you try with default parameter?
